batch/sql/estimated-current.sql

```diff
@@ -1306,6 +1338,85 @@ BEGIN
   COMMIT;
 END $$
 
+DROP PROCEDURE IF EXISTS cancel_job_group $$
+CREATE PROCEDURE cancel_job_group(
+  IN in_batch_id VARCHAR(100),
+  IN in_job_group_id INT
+)
+BEGIN
+  DECLARE cur_user VARCHAR(100);
+  DECLARE cur_batch_state VARCHAR(40);
+  DECLARE cur_cancelled BOOLEAN;
+  DECLARE cur_n_cancelled_ready_jobs INT;
+  DECLARE cur_cancelled_ready_cores_mcpu BIGINT;
+  DECLARE cur_n_cancelled_running_jobs INT;
+  DECLARE cur_cancelled_running_cores_mcpu BIGINT;
+  DECLARE cur_n_n_cancelled_creating_jobs INT;
+
+  START TRANSACTION;
+
+  SELECT user, `state` INTO cur_user, cur_batch_state FROM batches
+  WHERE id = in_batch_id
+  FOR UPDATE;
+
+  SET cur_cancelled = EXISTS (SELECT TRUE
+                              FROM job_groups_cancelled
+                              WHERE id = in_batch_id AND job_group_id = in_job_group_id
+                              FOR UPDATE);
+
+  IF cur_batch_state = 'running' AND NOT cur_cancelled THEN
```

Comment:
Hmm. What is the purpose of the batch state column in our new era?

If the batch state is running then it's *possible* that this job group has running jobs, but not guaranteed. If there are no running jobs, I guess you could skip all this bookkeeping? 

What are the other batch states? Is there any harm in removing that if condition? It seems generally good to simplify predicates when possible.

Reply:
The other batch state (other than the backwards compatibility issue we ran into with "open") is "complete". I don't think we need those conditions at all unless we want to save the extra operations if it's already cancelled. That condition is from the original code Cotton wrote 4 years ago.

Reply:
My inclination is to remove the condition. The state seems almost vestigial at this point. All the resource tables should accurately reflect the what could need to be cancelled. Seems better to just rely on that being accurate.